src/Apps/Order/Routes/Respond/index.tsx

```diff
@@ -80,7 +93,62 @@ export class RespondRoute extends Component<RespondProps, RespondState> {
     })
   }
 
-  onMutationError(errors, errorModalTitle?, errorModalMessage?) {
+  createCounterOffer(price: number) {
+    return new Promise((resolve, reject) =>
+      commitMutation<RespondCounterOfferMutation>(
+        this.props.relay.environment,
+        {
+          mutation: graphql`
+            mutation RespondCounterOfferMutation(
+              $input: buyerCounterOfferInput!
+            ) {
+              ecommerceBuyerCounterOffer(input: $input) {
+                orderOrError {
+                  ... on OrderWithMutationSuccess {
+                    order {
+                      ...Respond_order
+                    }
+                  }
+                  ... on OrderWithMutationFailure {
+                    error {
+                      type
+                      code
+                      data
+                    }
+                  }
+                }
+              }
+            }
+          `,
+          variables: {
+            input: {
+              offerId: this.props.order.lastOffer.id,
+              offerPrice: {
+                amount: price,
+                currencyCode: "USD",
+              },
+            },
+          },
+          onCompleted: result => {
+            const orderOrError = result.ecommerceBuyerCounterOffer.orderOrError
+            if (orderOrError.error) {
+              reject(
+                new ErrorWithMetadata(
+                  orderOrError.error.code,
+                  orderOrError.error
+                )
+              )
+            } else {
+              resolve(orderOrError.order)
+            }
+          },
+          onError: reject,
```

Comment:
We should make a note to adopt this pattern elsewhere, the `resolve`/`reject` is 👍.

Reply:
I decided to try promisifying this as a test ahead of our 'refactor to remove duplication' work, and yeah it feels good 💆